diktat-rules/pom.xml

```diff
@@ -161,6 +161,73 @@
                     </arguments>
                 </configuration>
             </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <executions>
+                    <!-- Don't produce the JAR during the `package` phase. -->
+                    <execution>
+                        <id>default-jar</id>
+                        <phase>none</phase>
+                    </execution>
+                    <!-- Instead, create it immediately before the `test` phase
+                         so that the shade plug-in may use it. -->
+                    <execution>
+                        <id>fat-jar-for-smoke-tests</id>
+                        <phase>process-test-classes</phase>
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <!-- Create a fat JAR to be used in smoke tests, removing the
+                 circular dependency between `diktat-rules` and `diktat-ruleset`. -->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
```

Comment:
Why do we need a different plugin here? In `diktat-ruleset` we use `assembly-plugin`.

Reply:
In `diktat-ruleset`, we use the `assmebly-plugin` with the pre-configured assembly profile, and the resulting fat JAR is the **only** artifact that module produces (as you remember, the execution of `jar-plugin` is suppressed).

In `diktat-rules`, the situation is different: we still need the `jar-plugin` to run and produce a regular JAR, and we also need the fat JAR to be **excluded** from the list of module artifacts (so that this fat JAR here is not installed/deployed/used as a dependency).

That's why the `assembly-plugin` doesn't seem suitable here (not mentioning it would reauire a separate XML configuration in an external XML file).

Basically, the task was a bit more complex than just copying a plug-in configuration from one module to another.